src/MariaResult.cpp

```diff
@@ -34,8 +34,8 @@ void MariaResult::autocommit() {
   maria_conn->autocommit();
 }
 
-MariaResult* MariaResult::create_and_send_query(MariaConnectionPtr con, const std::string& sql) {
-  std::auto_ptr<MariaResult> res(new MariaResultPrep(con));
+MariaResult* MariaResult::create_and_send_query(MariaConnectionPtr con, const std::string& sql, bool is_stmnt) {
+  std::auto_ptr<MariaResult> res(new MariaResultPrep(con, is_stmnt));
```

Comment:
I have yet to encounter a query (with `!is_statement`) that doesn't work with the prepared statement protocol, but if this happens we may need to pass `is_statement` to the `MariaResultSimple()` constructor below. We don't need it now, but a comment to that effect might be worthwhile adding.

Reply:
I added a comment, although I have to admit, I do not understand what for. But if it's ok for you like this, it's ok for me 😉.